src/test/java/iudx/aaa/server/policy/CreatePolicyNotificationTest.java

```diff
@@ -368,8 +477,68 @@ void failNotifAlreadyExists(VertxTestContext testContext) {
     // in the .succeeding method, create the exact same thing (i.e. repeat the createNotification
     // call)
     // assert that you get a 409 + whatever else
-    testContext.completeNow();
-  }
 
+      String resourceGroup = url + "/da39a3ee5e6b4b0d3255bfef95601890afd80709/" + "rs." + url + "/"
+              + RandomStringUtils.randomAlphabetic(10).toLowerCase();
+
+      Map<String, List<String>> catClientRequest = new HashMap<String, List<String>>();
+      catClientRequest.put(RES_GRP, List.of(resourceGroup));
+
+      JsonObject ownerJson = provider.result();
+      JsonObject consumerJson = consumer.result();
+
+      UUID consumerId = UUID.fromString(consumerJson.getString("userId"));
+      UUID ownerId = UUID.fromString(ownerJson.getString("userId"));
+      UUID itemIdInDb = UUID.randomUUID();
+      UUID resServerId = UUID.randomUUID();
+
+      /* Mocking CatalogueClient.checkReqItems */
+      Mockito.doAnswer(i -> {
+          Map<String, List<String>> req = i.getArgument(0);
+          String catId = req.get(RES_GRP).get(0);
+          ResourceObj obj = new ResourceObj(RES_GRP, itemIdInDb, catId, ownerId, resServerId, null);
+
+          Map<String, ResourceObj> resp = new HashMap<String, ResourceObj>();
+          resp.put(catId, obj);
+          return Future.succeededFuture(resp);
+      }).when(catalogueClient).checkReqItems(catClientRequest);
+
+      /* Mocking RegistrationService.getUserDetails */
+      JsonObject ownerDetails = new JsonObject().put("email", ownerJson.getString("email"))
+              .put("name", new JsonObject().put("firstName", ownerJson.getString("firstName"))
+                      .put("lastName", ownerJson.getString("lastName")));
+
+      JsonObject consumerDetails = new JsonObject().put("email", consumerJson.getString("email"))
+              .put("name", new JsonObject().put("firstName", consumerJson.getString("firstName"))
+                      .put("lastName", consumerJson.getString("lastName")));
+      JsonObject userDetailsResp = new JsonObject().put(consumerId.toString(), consumerDetails)
+              .put(ownerId.toString(), ownerDetails);
+
+      mockRegistrationFactory.setResponse(userDetailsResp);
+
+      User user = new UserBuilder().keycloakId(consumerJson.getString("keycloakId"))
+              .userId(consumerJson.getString("userId"))
+              .name(consumerJson.getString("firstName"), consumerJson.getString("lastName"))
+              .roles(List.of(Roles.CONSUMER)).build();
+
+      JsonArray req = new JsonArray()
+              .add(new JsonObject().put("itemId", resourceGroup).put("itemType", "resource_group")
+                      .put("expiryDuration", "P1Y").put("constraints", new JsonObject()));
+
+      List<CreatePolicyNotification> request = CreatePolicyNotification.jsonArrayToList(req);
+
+      policyService.createPolicyNotification(request, user,
```

Comment:
Here, we need to first make a successful create request, and then make the exact same request after we complete the previous request- making the same request should result in a 409. 

See this test in the in the CreateApdTest to understand how to test this flow. https://github.com/datakaveri/iudx-aaa-server/blob/df67dbca11aa6a29563cbb2a04e273914f3a7ddd/src/test/java/iudx/aaa/server/apd/CreateApdTest.java#L354-L372

Reply:
Sorry, I did not read your test completely - did not notice the succeeding.